src/main/kotlin/org/rust/lang/core/macros/MacroExpander.kt

```diff
@@ -204,19 +209,39 @@ class MacroExpander(val project: Project) {
                 )
             }
         }
+}
 
-    /** Returns path from [context] to [def]'s crate */
-    private fun getPathToCrate(context: RsElement, def: RsMacro): String {
-        val externCrateMod = def.crateRoot ?: return ""
-        val contextCrate = context.crateRoot ?: return ""
-        if (externCrateMod == contextCrate) return ""
-        val externCrateItem = contextCrate.childrenOfType<RsExternCrateItem>()
-            .find { it.reference.resolve() == externCrateMod } ?: return ""
-        val name = externCrateItem.nameWithAlias
-        return "::$name"
-    }
+/**
+ * Returns (synthetic) path from [call] to [def]'s crate
+ * We can't just expand `$crate` to something like `::crate_name` because
+ * rustc always threat result of this expansion as a single identifier. E.g.
```

Comment:
Let's explain it more detailed.
It's not clear that we can pass `$crate` as an identifier but it's the main reason why you have to do such hacks.

Reply:
Also, maybe use a better verb than `threat` here?